File: Data Structures & Algorithms/pow-x-n/submission-1.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if (n == 0) return 1.0;
        double u = myPow(x, n/2);
        u = u*u;
        if (n < 0) return 1/myPow(x,-n);
        else if (n%2==1) return u * x;
        else return u;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        int l = 0, r = nums.length - 1;
        int pivot = nums.length;

        k = nums.length - k;

        while (pivot != k) {
            pivot = partition(nums, l, r);

            if (pivot < k) {
                l = pivot + 1;
            } else {
                r = pivot - 1;
            }
        }

        return nums[pivot];
    }

    private int partition(int[] nums, int l, int r) {
        int pivot = nums[r];
        int pivotIndex = l;

        while (l < r) {
            if (nums[l] <= pivot) {
                swap(nums, l, pivotIndex++);
            }

            l++;
        }

        swap(nums, pivotIndex, r);
        return pivotIndex;
    }

    void swap(int[] nums, int i, int j) {
        int tmp = nums[i];
        nums[i] = nums[j];
        nums[j] = tmp;
    }
}
